src/attached_probe.cpp

```diff
@@ -524,78 +488,68 @@ bool AttachedProbe::resolve_offset_uprobe(bool safe_mode, bool has_multiple_aps)
-static std::string find_vmlinux(const struct vmlinux_location *locs,
-                                struct symbol &sym)
-{
-  struct bcc_symbol_option option = {};
-  option.use_debug_file = 0;
-  option.use_symbol_type = BCC_SYM_ALL_TYPES ^ (1 << STT_NOTYPE);
-  struct utsname buf;
-
-  uname(&buf);
-
-  for (int i = 0; locs[i].path; i++) {
-    if (locs[i].raw)
-      continue; // This file is for BTF. skip
-    char path[PATH_MAX + 1];
-    snprintf(path, PATH_MAX, locs[i].path, buf.release);
-    if (access(path, R_OK))
-      continue;
-    bcc_elf_foreach_sym(path, sym_name_cb, &option, &sym);
-    if (sym.start) {
-      LOG(V1) << "vmlinux: using " << path;
-      return path;
-    }
-  }
-
-  return "";
-}
-
 void AttachedProbe::resolve_offset_kprobe(bool safe_mode)
 {
-  struct symbol sym = {};
-  std::string &symbol = probe_.attach_point;
-  uint64_t func_offset = probe_.func_offset;
-  offset_ = func_offset;
-
 #ifndef HAVE_UNSAFE_PROBE
   safe_mode = true;
 #endif
 
-  if (func_offset == 0)
+  offset_ = probe_.func_offset;
+
+  // If we are using only the symbol, we don't need to check the offset.
+  bool is_symbol_kprobe = !probe_.attach_point.empty();
+  if (is_symbol_kprobe && probe_.func_offset == 0)
     return;
 
-  sym.name = symbol;
-  const struct vmlinux_location *locs = vmlinux_locs;
-  struct vmlinux_location locs_env[] = {
-    { nullptr, false },
-    { nullptr, false },
-  };
-  char *env_path = std::getenv("BPFTRACE_VMLINUX");
-  if (env_path) {
-    locs_env[0].path = env_path;
-    locs = locs_env;
-  }
+  // Setup the symbol to resolve, either using the address or the name.
+  struct symbol sym = {};
+  if (is_symbol_kprobe)
+    sym.name = probe_.attach_point;
+  else
+    sym.address = probe_.address;
 
-  std::string path = find_vmlinux(locs, sym);
-  if (path.empty()) {
-    LOG(V1) << "Could not resolve symbol " << symbol
+  auto path = find_vmlinux(&sym);
+  if (!path.has_value()) {
+    if (probe_.attach_point.empty())
+      throw FatalUserException("Could not resolve address: " +
+                               std::to_string(probe_.address));
+
+    LOG(V1) << "Could not resolve symbol " << probe_.attach_point
             << ". Skipping usermode offset checking.";
     LOG(V1) << "The kernel will verify the safety of the location but "
                "will also allow the offset to be in a different symbol.";
-
     return;
   }
 
-  if (func_offset >= sym.size)
-    throw FatalUserException("Offset outside the function bounds ('" + symbol +
-                             "' size is " + std::to_string(sym.size) + ")");
+  // Populate probe_ fields according to the resolved symbol.
+  if (is_symbol_kprobe) {
+    probe_.address = sym.start + probe_.func_offset;
+  } else {
+    probe_.attach_point = std::move(sym.name);
+    if (__builtin_sub_overflow(probe_.address, sym.start, &probe_.func_offset))
+      throw FatalUserException("Offset before the function bounds ('" +
```

Comment:
Maybe this is an opportunity to utilize the missing_probes config to either throw, log a warning, or just ignore and skip the probe.